queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/shares.py

```diff
@@ -175,3 +167,70 @@ def create_share_class(share_class_info: dict) -> ShareClass:
         share_class.series.append(share_series)
 
     return share_class
+
+
+def update_business_share_class(share_classes: list, business: Business, exclusion_entries: list):
+    """Update existing ones in both db if they are present in json."""
+    share_class_db_ids = [item.id for item in business.share_classes]
+
+    inclusion_entries = []
+    for share_class_info in share_classes:
+        if share_class_info.get('id') in share_class_db_ids:
+            share_class = ShareClass.find_by_share_class_id(share_class_info.get('id'))
+            if share_class:
+                update_share_class(share_class, share_class_info)
+                inclusion_entries.append(share_class)
+            else:
+                exclusion_entries.append(share_class_info)
+        else:
+            exclusion_entries.append(share_class_info)
+
+    business.share_classes = inclusion_entries
+
+
+def update_share_class(share_class: ShareClass, share_class_info: dict):
+    """Update share class instance in db."""
+    share_class.name = share_class_info.get('name')
+    share_class.priority = share_class_info.get('priority')
+    share_class.max_share_flag = share_class_info.get('hasMaximumShares')
+    share_class.max_shares = share_class_info.get('maxNumberOfShares', None)
+    share_class.par_value_flag = share_class_info.get('hasParValue')
+    share_class.par_value = share_class_info.get('parValue', None)
+    share_class.currency = share_class_info.get('currency', None)
+    share_class.special_rights_flag = share_class_info.get('hasRightsOrRestrictions')
+
+    # array of ids for share series instance from db
+    share_class_series_ids = []
+    if len(share_class.series) > 0:
+        share_class_series_ids = [series.id for series in share_class.series]
+
+    # update existing series in db and create new series if not exist
+    for series_info in share_class_info.get('series'):
+        series_id = series_info.get('id')
+        if series_id in share_class_series_ids:
+            series_index = share_class_series_ids.index(series_id)
+            series = share_class.series[series_index]
+            update_share_series(series_info, series)
```

Comment:
Does this cover removing a share series from a share class

Reply:
No, it does not. Do you also want it to remove the existing share series from the share class?

Reply:
I  think you can remove a share series from an existing share class. It is the same logic you wrote for share class

Reply:
Alright. Thanks for clarifying this. It's now implemented.